commit ac9937b826c86256fdf571b11c92cec171499872
Author: Jonas Vautherin <jonas.vautherin@gmail.com>
Date:   Thu Sep 14 11:51:02 2017 +0200

    sitl: add possibility to not run gazebo when running make

diff --git a/Tools/sitl_run.sh b/Tools/sitl_run.sh
index 28e8c1bc5..d0cb7aa29 100755
--- a/Tools/sitl_run.sh
+++ b/Tools/sitl_run.sh
@@ -95,20 +95,23 @@ elif [ "$program" == "gazebo" ] && [ ! -n "$no_sim" ]
 then
 	if [ -x "$(command -v gazebo)" ]
 	then
-		# Set the plugin path so Gazebo finds our model and sim
-		source $src_path/Tools/setup_gazebo.bash ${src_path} ${build_path}
-
-		gzserver --verbose ${src_path}/Tools/sitl_gazebo/worlds/${model}.world &
-		SIM_PID=`echo $!`
-
-		if [[ -n "$HEADLESS" ]]; then
-			echo "not running gazebo gui"
-		else
-			# gzserver needs to be running to avoid a race. Since the launch
-			# is putting it into the background we need to avoid it by backing off
-			sleep 3
-			nice -n 20 gzclient --verbose &
-			GUI_PID=`echo $!`
+		if  [[ -z "$DONT_RUN" ]]
+		then
+			# Set the plugin path so Gazebo finds our model and sim
+			source $src_path/Tools/setup_gazebo.bash ${src_path} ${build_path}
+
+			gzserver --verbose ${src_path}/Tools/sitl_gazebo/worlds/${model}.world &
+			SIM_PID=`echo $!`
+
+			if [[ -n "$HEADLESS" ]]; then
+				echo "not running gazebo gui"
+			else
+				# gzserver needs to be running to avoid a race. Since the launch
+				# is putting it into the background we need to avoid it by backing off
+				sleep 3
+				nice -n 20 gzclient --verbose &
+				GUI_PID=`echo $!`
+			fi
 		fi
 	else
 		echo "You need to have gazebo simulator installed!"
@@ -141,8 +144,11 @@ sitl_command="$sudo_enabled $sitl_bin $no_pxh $chroot_enabled $src_path $src_pat
 
 echo SITL COMMAND: $sitl_command
 
+if [[ -n "$DONT_RUN" ]]
+then
+    echo "Not running simulation (\$DONT_RUN is set)."
 # Start Java simulator
-if [ "$debugger" == "lldb" ]
+elif [ "$debugger" == "lldb" ]
 then
 	lldb -- $sitl_command
 elif [ "$debugger" == "gdb" ]
@@ -170,14 +176,17 @@ else
 	$sitl_command
 fi
 
-if [ "$program" == "jmavsim" ]
-then
-	pkill -9 -P $SIM_PID
-	kill -9 $SIM_PID
-elif [ "$program" == "gazebo" ]
+if [[ -z "$DONT_RUN" ]]
 then
-	kill -9 $SIM_PID
-	if [[ ! -n "$HEADLESS" ]]; then
-		kill -9 $GUI_PID
+	if [ "$program" == "jmavsim" ]
+	then
+		pkill -9 -P $SIM_PID
+		kill -9 $SIM_PID
+	elif [ "$program" == "gazebo" ]
+	then
+		kill -9 $SIM_PID
+		if [[ ! -n "$HEADLESS" ]]; then
+			kill -9 $GUI_PID
+		fi
 	fi
 fi
